common/scripted_effects/wc_scourge_invasion_effects.txt

```diff
@@ -66,4 +91,128 @@ scourge_offer_vassalization_effect = {
 			offer_vassalization_interaction_effect = yes
 		}
 	}
+}
+
+find_scourge_jailor_and_save_to_scope_effect = {
+	find_scourge_story_effect = yes
+	
+	scope:story = {
+		random_in_list = {
+			variable = nerzhul_jailors
+			
+			limit = { is_avaliable_scourge_jailor_trigger = yes }
+			
+			save_scope_as = scourge_jailor
+		}
+	}
+}
+
+hurt_scourge_jailor_consequences_effect = {
+	$HURT_JAILOR$ = { save_scope_as = hurt_jailor }
+	$SCOURGE$ = {
+		if = {
+			limit = {
+				any_scourge_jailor_is_avaliable_trigger = yes
+				has_relation_scourge_jailor = scope:hurt_jailor
+				NOT = {
+					scope:killer = {
+						any_secret = {
+							secret_target = scope:hurt_jailor
+							secret_type = secret_murder
+						}
+					}
+				}
+			}
+			custom_description = {
+				text = imprison_scourge_jailor_consequences_tooltip
+				
+				trigger_event = { id = WCCSC.15 days = 3 }
+			}
+		}
+	}
+}
+
+end_scourge_effect = {
+	custom_description = {
+		text = end_scourge_tooltip
+		
+		find_scourge_story_effect = yes
+		
+		on_end_scourge_effect = yes
+		if = {
+			limit = { exists = scope:story }
+			scope:story = { end_story = yes }
+		}
+	}
+}
+on_end_scourge_effect = {
+	if = {
+		limit = { exists = title:e_scourge.holder }
+		shatter_realm_effect = { WINNER = title:e_scourge.holder LOSER = title:e_scourge.holder }
+	}
+}
+
+set_scourge_jailor_relations_effect = {
+	if = {
+		limit = { exists = var:legions_puppet }
+		every_in_list = {
+			variable = nerzhul_jailors
+			limit = {
+				is_alive = yes
+				NOT = { has_relation_scourge_damned = prev.story_owner }
+			}
+			set_relation_scourge_damned = prev.story_owner
+		}
+	}
+}
+
+fire_jailors_unavailable_event_effect = {
+	trigger_event = WCCSC.30
+}
+stops_being_legions_puppet_effect = {
+	custom_description = {
+		text = stops_being_legions_puppet_tooltip
+		
+		find_scourge_story_effect = yes
+		
+		scope:story = {
+			if = {
+				limit = { exists = var:legions_puppet }
+				remove_variable = legions_puppet
+			}
+		}
+		faith:death_god = {
+			if = {
+				limit = { has_doctrine = special_doctrine_legions_puppet }
+				remove_doctrine = special_doctrine_legions_puppet
+			}	
+		}
+		faith = {
+			if = {
+				limit = { has_doctrine = special_doctrine_legions_puppet }
+				remove_doctrine = special_doctrine_legions_puppet
+			}
+		}
```

Comment:
> It will fail to remove doctrine from one of the faiths if you create one and then convert back to the old one
> 
> Check how ecumenism is removed in `mend_great_schism_scripted_effect`

Changed in e2e35f591ee1c171dc12241fc19daacff8326374.